receiver/prometheusreceiver/internal/metadata.go

```diff
@@ -45,7 +45,7 @@ func (t *mService) Get(job, instance string) (MetadataCache, error) {
 
 	// from the same targetGroup, instance is not going to be duplicated
 	for _, target := range targetGroup {
-		if target.DiscoveredLabels().Get(model.AddressLabel) == instance {
+		if target.Labels().Get(model.InstanceLabel) == instance {
```

Comment:
Do we also need to switch from `DiscoveredLabels()` to `Labels()` in https://github.com/open-telemetry/opentelemetry-collector/blob/c0dd66ef47bd94e7578a874583903059a2fe7925/receiver/prometheusreceiver/internal/metadata.go#L65-L67 ?

Reply:
Currently SharedLabels is only used to get scheme. Can a scheme be renamed/dropped? not sure. Anyway if labels can be changed then we are suppose to use changed labels then it makes sense to replace here with `Labels()` as well. If there is a need to query Discovered labels in future for a specific reason then just add another interface to query discovered labels.

Reply:
AFAIK schema may change, though not sure if it is required for prometheusreceiver to work

Reply:
I would switch to `Labels()` there for consistency then.